services/ui_backend_service/data/cache/card_cache_manager.py

```diff
@@ -0,0 +1,289 @@
+import time
+from subprocess import Popen
+import os
+import sys
+from services.utils import logging
+import uuid
+from asyncio.subprocess import Process
+import asyncio
+from .card_cache_service import CardCache, cleanup_non_running_caches
+import contextlib
+
+
+_PARENT_DIR = os.path.dirname(__file__)
+PATH_TO_CACHE_SERVICE = os.path.join(_PARENT_DIR, "card_cache_service.py")
+
+DATA_UPDATE_POLLING_FREQUENCY = os.environ.get(
+    "CARD_CACHE_DATA_UPDATE_POLLING_FREQUENCY", 0.3
+)
+CARD_UPDATE_POLLING_FREQUENCY = os.environ.get(
+    "CARD_CACHE_CARD_UPDATE_POLLING_FREQUENCY", 2
+)
+CARD_LIST_POLLING_FREQUENCY = os.environ.get(
+    "CARD_CACHE_CARD_LIST_POLLING_FREQUENCY", 2
+)
+CARD_CACHE_PROCESS_MAX_UPTIME = os.environ.get(
+    "CARD_CACHE_PROCESS_MAX_UPTIME", 3 * 60  # 3 minutes
+)
+CARD_CACHE_PROCESS_NO_CARD_WAIT_TIME = os.environ.get(
+    "CARD_CACHE_PROCESS_NO_CARD_WAIT_TIME", 4  # 4 seconds
+)
+DEFAULT_CACHE_STORAGE_PATH = "/tmp"
+CACHE_STORAGE_PATH = os.environ.get(
+    "CARD_CACHE_STORAGE_PATH", DEFAULT_CACHE_STORAGE_PATH
+)
+CACHE_SERVICE_LOG_STORAGE_ROOT = os.environ.get("CACHE_SERVICE_LOG_STORAGE_ROOT", None)
+
+CARD_API_HTML_WAIT_TIME = float(os.environ.get("CARD_API_HTML_WAIT_TIME", 5))
+
+
+async def _get_latest_return_code(process: Process):
+    with contextlib.suppress(asyncio.TimeoutError):
+        await asyncio.wait_for(process.wait(), 1e-6)
```

Comment:
just some notes for clarity.. so the combination of `wait_for` and async `process.wait()` with a minimal timeout in effect acts in a similar way as `process.poll()` from the synchronous implementation?